### DISCUSSION TITLE
panoptic segmentation

### DISCUSSION THREAD

**Discussion by aws-ahmad** [yoshitomo-matsubara/torchdistill#313] [Q&A] [answered]
Hi I have a mask2former dectron2 model backbone swin large with one class and panoptic segmentation.
I want to use your library to distillate this model to  student model using backbone swin tiny.
What do you think I need to do is to modify your coco2017 two stage yaml file to be suitable with my configuration.
and what i have to modify in  example where you use  semantic segmentation to to use panoptic segmentation.

**Comment by yoshitomo-matsubara** [comment]
Hi @aws-ahmad 

See related discussions at first
e.g.,
- https://github.com/yoshitomo-matsubara/torchdistill/discussions?discussions_q=is%3Aopen+object+detection
- https://github.com/yoshitomo-matsubara/torchdistill/issues/117

Also I need more details to answer your questions.
1. What is your student model?
2. What knowledge distillation method do you want to use?
3. I am not sure what you mean by "a mask2former dectron2 model backbone swin large **with one class and panoptic segmentation.**"

**Reply by aws-ahmad** [reply]
student mode: backbon swin transformer tiny
teacher mode: backbon swin transformer large
"a mask2former dectron2 model backbone swin large with one class and panoptic segmentation." I mean that i want to use panoptic segmitation.  Teacher Model Predicts One Class

**Reply by yoshitomo-matsubara** [reply]
2. What knowledge distillation method do you want to use?

**Reply by aws-ahmad** [reply]
At first i would like to use Response-based knowledge. Offline distillation
What do you think about it?

**Reply by yoshitomo-matsubara** [reply]
I am not aware of the method, response-based knowledge offline distillation. 
Where can we find the paper? And how different is the approach from other KD methods?

**Reply by aws-ahmad** [reply]
Response-based knowledge: the main idea of response-based knowledge distillation is to directly transfer the last output layer neural response of the teacher model. The hypothesis is that the student model will learn to mimic the predictions of the teacher model.  This can be achieved by using a loss function, termed the distillation loss, that captures the difference between the logits of the student and the teacher model respectively. As this loss is minimized over training, the student model will become better at making the same predictions as the teacher.
Offline distillation is the most common method, where a pre-trained teacher model is used to guide the student model. In this scheme, the teacher model is first pre-trained on a training dataset, and then knowledge from the teacher model is distilled to train the student model. Given the recent advances in deep learning, a wide variety of pre-trained neural network models are openly available that can serve as the teacher depending on the use case. Offline distillation is an established technique in deep learning and easier to implement.

it isn't a paper, it is just one type of KD

**Reply by yoshitomo-matsubara** [reply]
It looks like those are not your words, and you don't know what method you want to test.

So I'd suggest that you start with some existing configs (e.g., [KTAAD for DeepLabv3 w/ ResNet50 and DeepLab w/MobileNet v3 large as teacher and student models](https://github.com/yoshitomo-matsubara/torchdistill/blob/main/configs/legacy/sample/coco2017/multi_stage/ktaad/deeplabv3_mobilenet_v3_large_from_deeplabv3_resnet50.yaml)) and change models, modules to extract its input/output, other hyperparamters.

**Reply by aws-ahmad** [reply]
Yes, for sure, I just wanted to show you what i want to do.
I found  on the Internet
https://deci.ai/blog/knowledge-distillation-introduction/
there are several methods here, 
Where I can found a tutorial about torchdistill?

**Reply by aws-ahmad** [reply]
Where I can find a tutorial about torchdistill?

**Reply by yoshitomo-matsubara** [reply, marked as answer]
Documentation is a work in progress.
Learn from examples (sample configs) and other discussions as there are already some similar configs/questions
